DESIGN.md

```diff
@@ -74,4 +74,35 @@ It is "hot" because consumers subscribing to it does not cause side-effects, or
 
 A type representing work that can be cancelled or disposed.
 
+### Behavior
+
+##### Creation
+
+Creation of a stream falls into the following use cases, all of which should be catered to in API design.
+
+- async, hot, push (ie. system or user events)
+- async, cold, push (ie. events resulting from remote system via network connection)
+- sync, cold, pull (ie. iterable, file, range)
+- async, cold, pull (ie. RPC/REST network call, cross-thread queue draining)
```

Comment:
>RPC/REST network call

sounds like "initiating network requests" as opposed to "accepting network requests" which would be the `async, cold, push` pattern above. This has the option of canceling or optionally not parsing the response while the other push case is bound to the usage contract of not dropping/losing the request and producing a response.